ws-security-dom/src/main/java/org/apache/wss4j/dom/processor/EncryptedKeyProcessor.java

```diff
@@ -357,6 +364,106 @@ private static byte[] getAsymmetricDecryptedBytes(
+
+        try {
+            cipher.init(Cipher.UNWRAP_MODE, kek);
+            String keyAlgorithm = JCEMapper.translateURItoJCEID(encryptedKeyTransportMethod);
+            return cipher.unwrap(encryptedEphemeralKey, keyAlgorithm, Cipher.SECRET_KEY).getEncoded();
+        } catch (InvalidKeyException | NoSuchAlgorithmException ex) {
+            throw new WSSecurityException(WSSecurityException.ErrorCode.FAILED_CHECK, ex);
+        }
+    }
+
+    /**
+     * if keyInfo element contains AgreementMethod element then check if it is supported EC Diffie-Hellman key agreement algorithm
+     *
+     * @param keyInfoChildElement The KeyInfo child element
+     * @return true if AgreementMethod element is present and DH algorithm supported and false if AgreementMethod element is not present
+     * @throws WSSecurityException if AgreementMethod element is present but DH algorithm is not supported
+     */
+    private boolean isDiffieHellmanKeyWrap(Element keyInfoChildElement) throws WSSecurityException {
+        if (EncryptionConstants._TAG_AGREEMENTMETHOD.equals(keyInfoChildElement.getLocalName())
+                && WSConstants.ENC_NS.equals(keyInfoChildElement.getNamespaceURI())) {
+            String algorithmURI = keyInfoChildElement.getAttributeNS(null, "Algorithm");
+            // Only ECDH_ES is supported for AgreementMethod
+            if (!WSConstants.AGREEMENT_METHOD_ECDH_ES.equals(algorithmURI)) {
+                throw new WSSecurityException(
+                        WSSecurityException.ErrorCode.UNSUPPORTED_ALGORITHM,
+                        "unknownAlgorithm", new Object[]{algorithmURI});
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Parse keyInfo content to AgreementMethod object.
+     *
+     * @param keyInfoChildElement The KeyInfo child element containing AgreementMethod data.
+     * @return the {@link AgreementMethod} object.
+     * @throws WSSecurityException if AgreementMethod element is invalid.
+     */
+    private AgreementMethod getAgreementMethodFromElement(Element keyInfoChildElement) throws WSSecurityException {
+        try {
+            return new AgreementMethodImpl(keyInfoChildElement);
+        } catch (XMLSecurityException ex) {
+            throw new WSSecurityException(WSSecurityException.ErrorCode.INVALID_SECURITY, ex);
+        }
+    }
+
+    /**
+     * Get the RecipientKeyInfo child element from the AgreementMethod element.
+     *
+     * @param agreementMethod The AgreementMethod element
+     * @return the RecipientKeyInfo child element which contains the recipient's public key.
+     * @throws WSSecurityException if the agreementMethod is null or RecipientKeyInfo element can not be retrieved.
+     */
+    private Element getRecipientKeyInfoChildElement(AgreementMethod agreementMethod) throws WSSecurityException {
+        try {
+            RecipientKeyInfo recipientKeyInfo = agreementMethod.getRecipientKeyInfo();
```

Comment:
Isn't there a potential NPE on agreementMethod here as well?

Reply:
The way it is used no, because method 
(line 155) 
agreementMethod = getAgreementMethodFromElement(keyInfoChildElement);
always returns object or it throws an error.